cloudinit/sources/DataSourceAzure.py

```diff
@@ -208,6 +207,33 @@ def get_hv_netvsc_macs_normalized() -> List[str]:
     ]
 
 
+@azure_ds_telemetry_reporter
+def determine_device_driver_for_mac(mac: str) -> Optional[str]:
+    """Determine the device driver to match on, if any."""
+    interfaces = [
+        i for i in net.get_interfaces() if mac == normalize_mac_address(i[1])
+    ]
+    drivers = [i[2] for i in interfaces if i[2] not in BLACKLIST_DRIVERS]
```

Comment:
Driveby comment: Could we use the `blacklist_drivers` kwarg in `get_interfaces()` instead of manually filtering?

Reply:
Also I'd like to see `get_interfaces()` return a named tuple at some point so this is all more readable, but that's obviously not your problem or something that needs to be addressed in this PR.

Reply:
> Driveby nitpick: Could we use the `blacklist_drivers` kwarg in `get_interfaces()` instead of manually filtering?

I had done that originally but then I preferred the clarity in the unittest scenarios when mocking return data for get_interfaces().  But now that i've rounded out the scenarios I can go back and clean it up.

Reply:
Fixed over in #1672